Tests/SwiftParserTest/DeclarationTests.swift

```diff
@@ -891,6 +891,131 @@ final class DeclarationTests: ParserTestCase {
     )
   }
 
+  func testTypedThrows() {
+    assertParse(
+      "func test() throws(any Error) -> Int { }",
+      experimentalFeatures: [.typedThrows]
+    )
+
+    assertParse(
+      """
+      struct X {
+        init() throws(any Error) { }
+      }
+      """,
+      experimentalFeatures: [.typedThrows]
+    )
+
+    assertParse(
+      "func test() throws(MyError) 1️⃣async {}",
+      diagnostics: [
+        DiagnosticSpec(message: "'async' must precede 'throws'", fixIts: ["move 'async' in front of 'throws'"])
+      ],
+      fixedSource: "func test() async throws(MyError) {}",
+      experimentalFeatures: [.typedThrows]
+    )
+    assertParse(
+      "func test() throws 1️⃣async2️⃣(MyError) {}",
+      diagnostics: [
+        DiagnosticSpec(locationMarker: "1️⃣", message: "'async' must precede 'throws'", fixIts: ["move 'async' in front of 'throws'"]),
+        DiagnosticSpec(locationMarker: "2️⃣", message: "unexpected code '(MyError)' in function"),
+      ],
+      fixedSource: "func test() async throws (MyError) {}",
```

Comment:
@ahoppen are we missing removing `(MyError)` here because they end up intersecting?

Reply:
I think it's grabbing both the `async` and the unexpected nodes after the `async` and moving them forward.